pytab/core.py

```diff
@@ -300,6 +307,32 @@ def cli(
     system_info = hwi.get_system_info()
     click.echo(" success!")
 
+    # Logic for GPU Selection
+    gpus = system_info["gpu"]["gpu_elements"]
+    if len(gpus) > 1 and "gpu_id" not in globals():
+        click.echo("| Multiple GPU's detected")
+        for gpu in gpus:
+            click.echo(f"| {gpu["id"]}: {gpu["name"]} ")
+        gpu_id = click.prompt(
+            f"| please select a gpu id\n| Must be less than {len(gpus)}", type=int
+        )
+        if gpu_id > len(gpus):
+            click.echo("| GPU selection was not in range")
+            gpu_id = click.prompt(
+                f"| Please try again\n| Must be less than {len(gpus)}", type=int
+            )
+        gpu_ind = gpu_id - 1
+    elif "gpu_id" in globals() and len(gpus) > 1:
```

Comment:
Since this only triggers when the `--gpu` option is used, its best to NOT ask for correction here. Its better to just crash with an Error message, then to hold the Process for unlimited time.

Also the `globals` Part is recursive from above and needs to be changed.